labels.go

```diff
@@ -126,18 +126,17 @@ func Split(s string) []int {
 // The bool end is true when the end of the string has been reached.
 // Also see PrevLabel.
 func NextLabel(s string, offset int) (i int, end bool) {
-	quote := false
 	for i = offset; i < len(s)-1; i++ {
```

Comment:
What’s the performance like if you use `strings.Index(s[i:], ".")`?

Reply:
Will try. That will probably heavily depend on if the optimizer inlines Index.

Reply:
It's about 25% slower using IndexByte. I implemented two versions. The code is [here](https://github.com/jpicht/dns/tree/IndexByteLoop) and [here](https://github.com/jpicht/dns/tree/IndexByteRecursive).

Reply:
That’s disappointing. Oh well.